bee-network/src/config.rs

```diff
@@ -0,0 +1,145 @@
+}
+
+impl Config {
+    /// Creates a new config.
+    pub fn new(port: u16, local_id: LocalIdentity, peers_file_path: &str) -> Self {
+        let bind_addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port);
+        let manual_peer_config = ManualPeerConfig::from_file(&local_id, peers_file_path);
+
+        Self {
+            bind_addr,
+            local_id,
+            manual_peer_config,
+        }
+    }
+}
+
+/// Stores connection and other information about a manual peer.
+#[derive(Debug, Clone)]
+pub struct ManualPeerInfo {
+    /// The identity of the peer.
+    pub identity: Identity,
+    /// The address of the peer.
+    pub address: SocketAddr,
+    /// A human friendly identifier of the peer.
+    pub alias: String,
+    dialer: bool,
+}
+
+impl ManualPeerInfo {
+    /// Whether the peer is supposed to be the initiator of a connection.
+    pub fn is_dialer(&self) -> bool {
+        self.dialer
+    }
+}
+
+/// A YAML backed database for manual peers.
+#[derive(Clone)]
+pub struct ManualPeerConfig {
+    infos: HashMap<IpAddr, ManualPeerInfo>,
+}
+
+impl ManualPeerConfig {
+    /// Restores the database from its backing file.
+    pub fn from_file(local_id: &LocalIdentity, path: &str) -> Self {
+        let mut file = File::open(path).expect("error opening peers file");
+        let mut s = String::new();
+
+        file.read_to_string(&mut s).expect("error reading peers file");
+
+        let mut docs = YamlLoader::load_from_str(&s).expect("error parsing peers file");
+        assert_eq!(1, docs.len());
+
+        let doc = docs.remove(0);
+
+        assert!(doc.is_array(), "peers yaml is not an array");
+
+        let peers_config = doc.into_vec().unwrap();
+        let mut infos = HashMap::with_capacity(peers_config.len());
+
+        for hm in peers_config
+            .into_iter()
+            .map(|m| m.into_hash().expect("invalid yaml file"))
+        {
+            let public_key_str = hm.get(&Yaml::String("public_key".into())).unwrap().as_str().unwrap();
+            let public_key = util::from_public_key_string(public_key_str);
+
+            let dialer = public_key < local_id.public_key();
+
+            let identity = Identity::from_public_key(public_key);
+
+            let address = hm
+                .get(&Yaml::String("address".into()))
+                .unwrap()
+                .as_str()
+                .unwrap()
+                .parse::<SocketAddr>()
+                .unwrap();
+            let ip = address.ip();
+
+            let alias = hm
+                .get(&Yaml::String("alias".into()))
+                .unwrap()
+                .as_str()
+                .unwrap()
+                .to_string();
+
+            let peer_info = ManualPeerInfo {
+                identity,
+                address,
+                alias,
+                dialer,
+            };
+
+            if infos.contains_key(&ip) {
+                unimplemented!("multiple instances with same ip address");
```

Comment:
I think the notion of `unimplemented` is that we want to make the "user" aware that we currently do not intend to implement this (unless this becomes a requested feature in the future). If something is just not implemented due to "priorities" (but is considered to be in the needs-to-be-done before release), then I'ld use the `todo!` macro. Because I wanted to transport the context (that we currently do not intend to implement this "feature"), I think it makes sense, but if others lean more towards your pov I am happy to change.